src/components/dropdown.component.js

```diff
@@ -1,43 +1,93 @@
-import $ from 'jquery';
-
-export default class DropdownComponent {
-    constructor(query) {
-        this.query = query;
-
-        this.btn = $(`${query} .dropdown-btn`);
-        this.items = $(`${query} .dropdown-item`);
-        this.list = $(`${query} .dropdown-list`);
-        this.selectedItem = $(`${query} .selected`);
-
-        this.addDropdownClickEventListner();
-        this.addItemClickEventListner();
-    }
-
-    addDropdownClickEventListner() {
-        this.btn.on('click', () => {
-            this.list.toggle();
-        });
-    }
-
-    addItemClickEventListner() {
-        this.items.on('click', (event) => {
-            this.unselectItem();
-            this.selectItem(event.currentTarget);
-            this.list.toggle();
-        });
-    }
-
-    updateDropdown() {
-        this.btn.html(this.selectedItem.text());
-    }
-
-    unselectItem() {
-        this.items.removeClass('selected');
-    }
-
-    selectItem(item) {
-        this.selectedItem = $(item);
-        this.selectedItem.addClass('selected');
-        this.updateDropdown();
-    }
-}
+import $ from 'jquery';
+
+export default class DropdownComponent {
+    constructor(query, required = true) {
+        this.query = query;
+        this.required = required;
+
+        this.container = $(`${query} .dropdown-container`);
+        this.btn = $(`${query} .dropdown-btn`);
+        this.list = $(`${query} .dropdown-list`);
+        this.items = $(`${query} .dropdown-item`);
+        this.selectedItem = $(`${query} .selected`);
+
+        this.disabled = false;
+        this.valid = false;
+
+        this.defaultText = this.btn.text();
+        this.unselectedDefaultText = this.selectedItem.text();
+
+        this.addDropdownClickEventListener();
+        this.addItemClickEventListener();
+    }
+
+    setSelection(item) {

```

Comment:
It is not dealing with uppercase or lowercase letters
![image](https://user-images.githubusercontent.com/23269934/162782126-ac18c9d9-a073-426a-9629-a4e73eaf4f2c.png)

Reply:
I've add .toLowerCase() in both comparations.